Formula/re-flex.rb

```diff
@@ -1,23 +1,29 @@
 class ReFlex < Formula
+  env :std
   desc "The regex-centric, fast and flexible scanner generator for C++"
   homepage "https://www.genivia.com/doc/reflex/html"
-  url "https://github.com/Genivia/RE-flex/archive/v1.6.6.tar.gz"
-  sha256 "be98ae3b270ebec4261f5bc78b5bbbe02ce6afee080c175c9a6bb8f01e28eaee"
+  url "https://github.com/Genivia/RE-flex/archive/v1.6.7.tar.gz"
+  sha256 "526e66f7317cfaa9ea017ad0a83a5c2f8ae469f9c522eed989eb83f16342c72d"
 
   bottle do
     cellar :any_skip_relocation
-    sha256 "b3aaeca6a4d5b42f3bd117a63f33000e7997ac2613eb3724b7452c060c05980e" => :catalina
-    sha256 "e5159fd4b0064a475cec8065366482dbcc3335f899bf35446e71fb90cbed305e" => :mojave
-    sha256 "fa43c6264ba478c09aeffd748e5e8242b1ac678d4470d3b12187f9d7ec9f1628" => :high_sierra
+    sha256 "075657453a8dc116d425965a5ad8432b3b66b8b9cfd581eabdcaa0c595e5ebf1" => :catalina
+    sha256 "caf15bf38d93db054e1f90a9182802b7df688b3266dccd48ccd7550c7d7d50e7" => :mojave
+    sha256 "aeee136f831845272ba2e5b67aee5cbe6e93a6d4b77a9991f5bd0bda88f93e76" => :high_sierra
   end
 
-  depends_on "boost"
-  depends_on "pcre2"
+  depends_on "pcre2" => :recommended
+  depends_on "boost" => :optional
```

Comment:
`:recommended` and `:optional` aren't accepted in Homebrew/core anymore.

Reply:
Right... These are mentioned in the [Homebrew Cookbook](https://docs.brew.sh/Formula-Cookbook). It makes more sense to have Boost.Regex optional, since PCRE2 is preferred.

Reply:
https://docs.brew.sh/Formula-Cookbook#specifying-other-formulae-as-dependencies has a note
> Note: options are not allowed in Homebrew/homebrew-core as they are not tested by CI.

Reply:
Either the documentation should remove `:optional` or this should be resolved by giving the user installing with an optional feature a warning. This is in the interest of the user, we as developers are taking the blame anyway if this is an issue. In this case it is really up to the user to decide to install Boost. Also, installing all of Boost is an overkill. Only the Boost.Regex component is useful and optional.

Reply:
There's a ton of taps out in the world that still offer this feature, it just isn't supported in the main one. But you can remove the boost dependency if it's not needed.

Reply:
I'll remove the Boost dependency. The re-flex documentation explains that the use of Boost.Regex is optional. Same for PCRE2 for re-flex, but PCRE2 is quick to install by contrast to Boost. In fact, when I tested this on my Raspberry Pi 3B+, the RPi chocked on it for many hours and became unresponsive. So I turned it off. That isn't good. That's why I decided to make it optional. We don't want `brew install` to fail that easily. If there is a way to install a single component of Boost (e.g. Boost.Regex) then I'd like to know how.

Reply:
Source builds or ARM aren't the primary use case for homebrew though, almost all installs of boost will just download and extract the precompiled bottle that homebrew ships.

Reply:
Still, Boost is HUGE and best to make it optional.

Reply:
Oops, forgot that `:recommend` cannot be used either. Please be patient. The last push should be OK now.

Reply:
With the last push and tests succeeding now by removing the `env :std` and `ENV`, I tried to find the output produced by `./configure` to see if AVX or SSE2 are used, if at all. I cannot find this information. Any reasons I should have to worry that AVX and SSE2 are not enabled at all?